src/node_crypto.cc

```diff
@@ -4085,33 +4086,23 @@ static int Node_SignFinal(EVP_MD_CTX* mdctx, unsigned char* md,
   if (!EVP_DigestFinal_ex(mdctx, m, &m_len))
     return rv;
 
-  if (mdctx->digest->flags & EVP_MD_FLAG_PKEY_METHOD_SIGNATURE) {
-    size_t sltmp = static_cast<size_t>(EVP_PKEY_size(pkey));
-    pkctx = EVP_PKEY_CTX_new(pkey, nullptr);
-    if (pkctx == nullptr)
-      goto err;
-    if (EVP_PKEY_sign_init(pkctx) <= 0)
-      goto err;
-    if (!ApplyRSAOptions(pkey, pkctx, padding, pss_salt_len))
-      goto err;
-    if (EVP_PKEY_CTX_set_signature_md(pkctx, mdctx->digest) <= 0)
-      goto err;
-    if (EVP_PKEY_sign(pkctx, md, &sltmp, m, m_len) <= 0)
-      goto err;
-    *sig_len = sltmp;
-    rv = 1;
-   err:
-    EVP_PKEY_CTX_free(pkctx);
-    return rv;
-  }
-
-  if (mdctx->digest->sign == nullptr) {
-    EVPerr(EVP_F_EVP_SIGNFINAL, EVP_R_NO_SIGN_FUNCTION_CONFIGURED);
-    return 0;
-  }
-
-  return mdctx->digest->sign(mdctx->digest->type, m, m_len, md, sig_len,
```

Comment:
This is the same fix as in openssl-1.1.0 and `EVP_MD_FLAG_PKEY_METHOD_SIGNATURE` was removed in https://github.com/openssl/openssl/commit/7f572e958b13041056f377a62d3219633cfb1e8a. Is it sure we do not need this code path even in 1.0.2?

Reply:
Node is already assuming this codepath isn't needed in the verify half. This codepath is because, in 0.9.8, an `EVP_PKEY` didn't have `EVP_PKEY_sign` and `EVP_PKEY_verify`. Those were implemented in the signature `EVP_MD`s. In 1.0.0, an `EVP_PKEY` became less tied to `EVP_MD` and could sign and verify digests on their own.

Reply:
Thanks for your explanation about its history. I thought it might be safe for we do not have NoneWithRSA but I did not have the confidence to remove that code path. .